seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java

```diff
@@ -70,13 +82,39 @@ public List<Dataset<Row>> execute(List<Dataset<Row>> upstreamDataStreams) throws
         Dataset<Row> input = upstreamDataStreams.get(0);
         List<Dataset<Row>> result = new ArrayList<>();
         for (int i = 0; i < plugins.size(); i++) {
-            BaseSparkTransform transform = plugins.get(i);
-            Config pluginConfig = pluginConfigs.get(i);
-            Dataset<Row> stream = fromSourceTable(pluginConfig, sparkEnvironment).orElse(input);
-            input = transform.process(stream, sparkEnvironment);
-            registerInputTempView(pluginConfig, input);
-            result.add(input);
+            try {
+                SeaTunnelTransform<SeaTunnelRow> transform = plugins.get(i);
+                Config pluginConfig = pluginConfigs.get(i);
+                Dataset<Row> stream = fromSourceTable(pluginConfig, sparkEnvironment).orElse(input);
+                input = sparkTransform(transform, stream);
+                registerInputTempView(pluginConfig, input);
+                result.add(input);
+            } catch (Exception e) {
+                throw new TaskExecuteException(
+                    String.format("SeaTunnel transform task: %s execute error", plugins.get(i).getPluginName()), e);
+            }
         }
         return result;
     }
+
+    private Dataset<Row> sparkTransform(SeaTunnelTransform transform, Dataset<Row> stream) throws IOException {
+        SeaTunnelDataType<?> seaTunnelDataType = TypeConverterUtils.convert(stream.schema());
+        transform.setTypeInfo(seaTunnelDataType);
+        StructType structType = (StructType) TypeConverterUtils.convert(transform.getProducedType());
+        List<Row> outputRows = new ArrayList<>();
+        Iterator<Row> rowIterator = stream.toLocalIterator();
+        SeaTunnelRow seaTunnelRow;
+        while (rowIterator.hasNext()) {
+            Row row = rowIterator.next();
+            seaTunnelRow = new InternalRowConverter(seaTunnelDataType).reconvert(InternalRow.apply(row.toSeq()));
+            seaTunnelRow = (SeaTunnelRow) transform.map(seaTunnelRow);
+            InternalRow internalRow = new InternalRowConverter(transform.getProducedType()).convert(seaTunnelRow);
+            MutableValue[] mutableValues = ((SpecificInternalRow) internalRow).values();
+            outputRows.add(new GenericRowWithSchema(
+                Arrays.stream(mutableValues).map(MutableValue::boxed).toArray(),
+                structType));
+        }
+        return sparkEnvironment.getSparkSession().createDataFrame(outputRows, structType);
```

Comment:
maybe should use `Dataset<Row> stream = Dataset#map(...)` or `Dataset<Row> stream = Dataset#transform(...)`?

Reply:
The above method cannot be serialized correctly.
![image](https://user-images.githubusercontent.com/58908699/207356092-15d59d06-b058-49db-87df-c600d2b37535.png)


Do you think the `Encoders` are right？